app/models/declaration/TransportData.scala

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models.declaration
+import forms.declaration.TransportPayment
+import play.api.libs.json.Json
+
+case class TransportData(transportPayment: Option[TransportPayment] = None, containers: Seq[Container] = Seq.empty) {
+
+  def addOrUpdateContainer(updatedContainer: Container): Seq[Container] =
+    if (containers.isEmpty) {
+      Seq(updatedContainer)
+    } else if (!containers.exists(_.id == updatedContainer.id)) {
+      containers :+ updatedContainer
+    } else {
+      containers.map {
+        case container if updatedContainer.id == container.id => updatedContainer
+        case otherContainer                                   => otherContainer
+      }
+    }
+}
```

Comment:
I think it'll be better and easier to understand when second if `!containers.exists(_.id == updatedContainer.id)` will be without negation and contains the last logic, and last `else` will have logic for new containers.

You'll be able to not use another nested `if` in last case

Reply:
Method was copied over from a different class. but I've re-structured it